lightning/src/ln/router.rs

```diff
@@ -1049,6 +1109,48 @@ impl Router {
 
 		Err(LightningError{err: "Failed to find a path to the given destination", action: ErrorAction::IgnoreError})
 	}
+
+	/// Gets the list of announced channels/nodes, in random order. See EdgeDetails and VerticeDetails field documentation for
+	/// more information.
+	pub fn list_graph(&self) -> (Vec<EdgeDetails>, Vec<VerticeDetails>) {
```

Comment:
Forcing the user into Vecs kinda sucks, maybe return an iterator (which the user can convert to a Vec if they want)?

Reply:
Split in two helpers, dunno about Iterator, we don't have collection of ChannelInfo/NodeInfo on which to iterate on, do you suggest to return HashMap/Btree in which we already store them ?
Note we already return a Vec for ChannelManager::list_channels